libs/tangle-shared-ui/src/data/blueprints/usePendingServiceRequest.ts

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useMemo } from 'react';
+import { toPrimitiveServiceRequest } from './utils/toPrimitiveService';
+import { catchError, map, of } from 'rxjs';
+import useApiRx from '../../hooks/useApiRx';
+import { TangleError, TangleErrorCode } from '../../types/error';
+import { createPendingServiceRequests } from './utils/blueprintHelpers';
+import { SubstrateAddress } from '@tangle-network/ui-components/types/address';
+import { toPrimitiveBlueprint } from './utils/toPrimitiveBlueprint';
+
+export default function usePendingServiceRequest(
+  operatorAccountAddress: SubstrateAddress | null,
+) {
+  const { result: serviceRequestEntries, ...rest } = useApiRx(
+    useCallback(
+      (apiRx) => {
+        if (apiRx.query?.services?.serviceRequests === undefined) {
+          return new TangleError(TangleErrorCode.FEATURE_NOT_SUPPORTED);
+        }
+
+        if (!operatorAccountAddress) {
+          return of([]);
+        }
+
+        return apiRx.query.services.serviceRequests.entries().pipe(
+          map((entries) => entries),
+          catchError((error) => {
+            console.error('Error querying service requests:', error);
+            return of([]);
+          }),
+        );
+      },
+      [operatorAccountAddress],
+    ),
+  );
+
+  const primitiveServiceRequests = useMemo(() => {
+    if (!serviceRequestEntries) return [];
+
+    return serviceRequestEntries
+      .map(([requestId, serviceRequest]) =>
+        toPrimitiveServiceRequest(requestId[0], serviceRequest.unwrap()),
+      )
+      .filter((serviceRequest) =>
+        serviceRequest.operatorsWithApprovalState.some(
+          (operator) =>
+            operator.operator === operatorAccountAddress &&
+            operator.approvalStateStatus === 'Pending',
+        ),
+      );
+  }, [serviceRequestEntries, operatorAccountAddress]);
+
+  const blueprintIds = useMemo(() => {
+    return primitiveServiceRequests.map(
+      (serviceRequest) => serviceRequest.blueprint,
+    );
+  }, [primitiveServiceRequests]);
+
+  const { result: blueprints } = useApiRx(
```

Comment:
Combine the `loading` and `error` states with `...rest` and return the result as metadata for this hook.